ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -2434,6 +2236,7 @@ _WARN("Session space stuff is not implemented yet!")
         /* Is this a guard page? */
         if ((ProtectionCode & MM_PROTECT_SPECIAL) == MM_GUARDPAGE)
```

Comment:
Do guard pages have a zero PTE? This whole path seems useless.

Reply:
I'll have to check what NtAllocateVirtualMemory does.

Reply:
Actually, it looks like NtAllocateVirtualMemory allows this if you use MEM_COMMIT without specifying a base address.

Reply:
Wow, yet another of these awesome features: reserve and commit an entire region of guard pages :D

Reply:
Somewhere out there, there may be JIT compiler implementing garbage collection using this feature.